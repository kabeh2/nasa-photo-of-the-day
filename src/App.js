import React, { useState, useEffect } from "react";
import axios from "axios";
import moment from "moment";
import logo from "./assets/nasa-logo.png";
import "./App.scss";
import Info from "./components/info";
import Media from "./components/media";
import CircularIndeterminate from "./components/loader";

function App() {
  const [loading, setLoading] = useState(false);
  // const [imageData, setImageData] = useState([]);
  const [media, setMedia] = useState("");
  const [mediaHD, setMediaHD] = useState("");
  const [title, setTitle] = useState("");
  const [info, setInfo] = useState("");
  const [imgDate, setImgDate] = useState("");
  const [mediaType, setMediaType] = useState("");

  const [startDate, setStartDate] = useState(new Date());
  const [newDate, setNewDate] = useState("");

  const handleChange = date => {
    const a = moment(date).format("YYYY-MM-DD");

    setStartDate(date);
    setNewDate(a);
  };

  useEffect(() => {
    setLoading(true);
    const fetchData = async () => {
      try {
        const result = await axios.get("https://api.nasa.gov/planetary/apod", {
          params: {
            date: newDate,
            api_key: "n86JVNah2R4q3KSYXeHnpMhvqDX9adH4cRMGrRgR"
          }
        });
        setLoading(false);
        setMediaHD(result.data.hdurl);
        setMedia(result.data.url);
        setTitle(result.data.title);
        setImgDate(result.data.date);
        setInfo(result.data.explanation);
        // setImageData(result.data);
        setMediaType(result.data.media_type);
      } catch (error) {
        console.log("there was an error");
      }
    };

    fetchData();
  }, [newDate]);

  // useEffect(() => {
  //   console.log("image url: " + media);
  //   console.log("imageData: " + JSON.stringify(imageData));

  //   console.log("Date new 1: " + newDate);
  // });

  return (
    <div className="App">
      <div
        style={{
          position: "absolute",
          top: "0",
          left: "0",
          zIndex: "1",
          width: "100vw",
          height: "100vh"
        }}
      >
        {loading ? (
          <CircularIndeterminate />
        ) : (
          <Media
            media={media}
            mediaHD={mediaHD}
            mediaType={mediaType}
            title={title}
          />
        )}
        {/* <Media
          media={media}
          mediaHD={mediaHD}
          mediaType={mediaType}
          title={title}
        /> */}
      </div>
      <Info
        logo={logo}
        date={imgDate}
        title={title}
        info={info}
        handleChange={handleChange}
        startDate={startDate}
        mediaType={mediaType}
      />
    </div>
  );
}

export default App;
