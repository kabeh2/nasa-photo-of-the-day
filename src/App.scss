.App {
  text-align: center;
  height: 100vh;
  width: 100vw;

  display: flex;
  align-items: flex-end;
  font-family: montserrat, sans-serif;
}

.info-container {
  width: 100%;
  height: 30%;
  padding: 24px 5%;
  background: linear-gradient(transparent, #000);

  display: flex;
  /* justify-content: center; */
  align-items: flex-end;

  .logo {
    width: 100px;
    height: 84.722px;
  }

  .info-container--content {
    display: flex;
    align-items: center;

    .info {
      // display: flex;
      // flex-direction: column;
      // justify-content: center;

      margin-left: 24px;
      text-align: left;

      color: white;
      // font-family: montserrat, sans-serif;
      font-size: 12px;

      &_title {
        font-size: 16px;
        font-weight: bold;
        letter-spacing: 2px;
        text-transform: uppercase;
        color: #ec1c24;
      }

      &_date {
        font-size: 14px;
        text-transform: uppercase;
        color: #0b66b2;
      }
    }
  }
}

h1 {
  font-size: 24px;
  text-transform: uppercase;
  letter-spacing: 2px;
  color: #fff;
  margin: 0;
}
