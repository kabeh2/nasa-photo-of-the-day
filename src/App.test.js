import React from "react";
import ReactDOM from "react-dom";
import App from "./App";
import { act } from "react-dom/test-utils";
import { shallow } from "enzyme";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  act(() => {
    const div = document.createElement("div");
    ReactDOM.render(<App />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
});

describe("App component", () => {
  it("matches the snapshot", () => {
    act(() => {
      const tree = renderer.create(<App />).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
