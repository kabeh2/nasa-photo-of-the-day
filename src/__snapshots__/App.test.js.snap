// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component matches the snapshot 1`] = `
<div
  className="App"
>
  <div
    style={
      Object {
        "height": "100vh",
        "left": "0",
        "position": "absolute",
        "top": "0",
        "width": "100vw",
        "zIndex": "1",
      }
    }
  >
    <div>
      <iframe
        allow="autoplay; encrypted-media"
        allowFullScreen={true}
        frameBorder="0"
        name="youtube embed"
        src="?rel=0&amp;autoplay=1&amp;enable_js=1&amp;mute=0"
        style={
          Object {
            "height": "100%",
            "left": 0,
            "position": "absolute",
            "top": 0,
            "width": "100%",
          }
        }
        title=""
        type="text/html"
      />
    </div>
  </div>
  <div
    className="info-container"
  >
    <div
      className="info-container--content"
    >
      <img
        alt="Nasa Logo"
        className="logo"
        src="nasa-logo.png"
      />
      <div
        className="info"
      >
        <h1>
          NASA 
          
           of the Day
        </h1>
        <div
          className="info_title"
        >
          
        </div>
        <div
          className="react-datepicker-wrapper"
        >
          <div
            className="react-datepicker__input-container"
          >
            <input
              className="datePicker"
              disabled={false}
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              placeholder=""
              readOnly={false}
              type="text"
              value="Saturday July 27, 2019"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
